daemon/src/maker_cfd.rs

```diff
@@ -345,6 +355,21 @@ where
             })
             .await?;
 
+        if !self.connected_takers.insert(taker_id) {
+            anyhow::bail!("Taker already connected: {:?}", &taker_id);
+        }
+        self.connected_takers_feed_sender
+            .send(self.connected_takers.clone().into_iter().collect())?;
+
+        Ok(())
+    }
+
+    async fn handle_taker_removed(&mut self, taker_id: TakerId) -> Result<()> {
+        if !self.connected_takers.remove(&taker_id) {
+            anyhow::bail!("Removed unknown taker: {:?}", &taker_id);
```

Comment:
Same here, perhaps logging is fine here?
Would make the function idempotent which is good I think.

Reply:
converted to log.